spec.html

```diff
@@ -597,11 +644,26 @@ <h1>Tree Grammar</h1>
 // `FunctionExpression`,
 // `GeneratorExpression`,
 // `AsyncFunctionExpression`,
-interface FunctionExpression : Node {
+interface EagerFunctionExpression : Node {
+    attribute FunctionExpressionPrologue prologue;
+    attribute FunctionExpressionContents contents;
+};
+interface LazyFunctionExpression : Node {
+    attribute FunctionExpressionPrologue prologue;
+    [Lazy] attribute FunctionExpressionContents contents;
+};
+interface FunctionExpressionPrologue : Node {
+  attribute BindingIdentifier? name;
+};
+
+interface FunctionExpressionContents : Node {
   attribute boolean isAsync;
   attribute boolean isGenerator;
```

Comment:
Why are the `isAsync` and `isGenerator` attributes not part of the Prologue? From the spec's perspective, it is easier to have the function kind to be always eagerly available when allocating Function objects.